shared/src/main/scala/pl/metastack/metatime/Time.scala

```diff
@@ -92,5 +112,13 @@ object Time {
   }
 
   /** Current time */
-  def now(): Time = ???
+  def now() : Time = {
+    val currTimeInMilliSec : Long = System.currentTimeMillis()
+    new Time {
+      override val h: Int = (((currTimeInMilliSec/ 1000) / 3600) % 24).toInt
+      override val m: Int = (((currTimeInMilliSec/1000) / 60) % 60).toInt
+      override val s: Int = ((currTimeInMilliSec/1000) % 60).toInt
+      override val ms: Int = (currTimeInMilliSec/1000).toInt
```

Comment:
`ms` does not get calculated properly. `currTimeInMilliSec` is already in milliseconds. What you have to do is subtract the milliseconds of `h`, `m` and `s`. It would make sense to move the logic for constructing a `Time` object to a helper function. Then it can be tested more easily.